gcc/rust/typecheck/rust-hir-type-check-toplevel.h

```diff
@@ -152,12 +152,21 @@ class TypeCheckTopLevel : public TypeCheckBase
       {
 	for (auto &generic_param : function.get_generic_params ())
 	  {
-	    auto param_type
-	      = TypeResolveGenericParam::Resolve (generic_param.get ());
-	    context->insert_type (generic_param->get_mappings (), param_type);
-
-	    substitutions.push_back (
-	      TyTy::SubstitutionParamMapping (generic_param, param_type));
+	    switch (generic_param.get ()->get_kind ())
+	      {
+	      case HIR::GenericParam::GenericKind::LIFETIME:
+		// Skipping Lifetime completely until better handling.
+		break;
+		case HIR::GenericParam::GenericKind::TYPE: {
```

Comment:
Same here.

Reply:
I really like this I have some changes locally that could benifit from this enum.